File: J9Ch/src/S_5_Dynamic_Problem_I/Required_10/_397_Longest_Increasing_Continuous_Subsequence_Easy.cpp

class Solution {
public:
    /**
     * @param A an array of Integer
     * @return  an integer
     */
    int longestIncreasingContinuousSubsequence(vector<int>& A) {
        // Write your code here
        int max = 1, s = 1, l = 1;
        int len = A.size();
        if (len == 0)
            return 0;
        for (int i = 1; i < len; ++i) {
            if (A[i] > A[i-1])
                s += 1;
            else {
                if (s > max) max = s;
                s = 1;
            }

            if (A[i] < A[i-1])
                l += 1;
            else {
                if (l > max) max = l;
                l = 1;
            }
        }
        if (s > max) max = s;
        if (l > max) max = l;
        return max;
    }
};

File: J9Ch/src/S_5_Dynamic_Problem_I/Required_10/_397_Longest_Increasing_Continuous_Subsequence_Easy.py


class Solution:
    # @param {int[]} A an array of Integer
    # @return {int}  an integer
    def longestIncreasingContinuousSubsequence(self, A):
        return max(self.getLongest(A), self.getLongest(list(reversed(A))))

    def getLongest(self, A):
        length, longest = 0, 0
        for index, a in enumerate(A):
            if index == 0 or A[index] < A[index - 1]:
                length = 1
            else:
                length += 1
            longest = max(longest, length)
        return longest